common/src/java/com/zimbra/common/mime/InternetAddress.java

```diff
@@ -497,10 +497,40 @@ private void parse(byte[] content, int start, int length, boolean angle) {
             parse(content, start, length, true);
         } else {
             this.display = base != null ? base : comment == null ? null : comment.toString();
+            this.display = sanitizeString(this.display);
             this.email = address == null ? null : address.toString().trim();
         }
     }
 
+    private static String sanitizeString(String value) {
+        if (value != null) {
+            StringBuilder newString = new StringBuilder(value.length());
+            for (int offset = 0; offset < value.length();) {
+                int codePoint = value.codePointAt(offset);
+                offset += Character.charCount(codePoint);
+                // Replace invisible control characters and unused code points
+                switch (Character.getType(codePoint)) {
+                case Character.CONTROL: // \p{Cc}
```

Comment:
nit: per internallly doc'd coding standard (Zimbra_Server_Code_Style in engineering wiki) we should indent case one level from the switch statement.

Reply:
replying to @fatkudu - it might be worth while to use a more generic sanitizer although I'll add these caveats:
1. we'd probably need a custom 'policy' as the html sanitizer doesn't seem to ship by default with a sanitizer that would be generic for rfc822 data - we're not dealing with an HTML encoding issue here, but a failure to handle data robustly when displaying the email addresses (and/or when replying to the given address)
2. our sanitization is removing data that could be (really is) valid data per the spec, but just because the spec allows for it, we probably still don't want to allow it due to the potential for abuse like we're trying to address here